lib/ts/recipe/dev/recipe.ts

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import RecipeModule from "../../recipeModule";
+import { NormalisedAppinfo, APIHandled, RecipeListFunction } from "../../types";
+import { TypeInput, RecipeInterface, APIInterface } from "./types";
+import STError from "./error";
+
+import { STATUS } from "./constants";
+import NormalisedURLPath from "../../normalisedURLPath";
+import RecipeImplementation from "./recipeImplementation";
+import APIImplementation from "./api/implementation";
+import { Querier } from "../../querier";
+import { BaseRequest, BaseResponse } from "../../framework";
+import healthCheckAPI from "./api/healthCheck";
+
+export default class Recipe extends RecipeModule {
+    private static instance: Recipe | undefined = undefined;
+    static RECIPE_ID = "dev";
+
+    config: TypeInput;
+
+    recipeInterfaceImpl: RecipeInterface;
+
+    apiImpl: APIInterface;
+
+    isInServerlessEnv: boolean;
+
+    constructor(recipeId: string, appInfo: NormalisedAppinfo, isInServerlessEnv: boolean, config: TypeInput) {
+        super(recipeId, appInfo);
+        this.config = config;
+        this.isInServerlessEnv = isInServerlessEnv;
+
+        this.recipeInterfaceImpl = new RecipeImplementation(Querier.getNewInstanceOrThrowError(recipeId));
+        this.apiImpl = new APIImplementation();
+    }
+
+    static init(config: TypeInput): RecipeListFunction {
+        return (appInfo, isInServerlessEnv) => {
+            if (Recipe.instance === undefined) {
+                Recipe.instance = new Recipe(Recipe.RECIPE_ID, appInfo, isInServerlessEnv, config);
+                return Recipe.instance;
+            } else {
+                throw new Error("Dev recipe has already been initialised. Please check your code for bugs.");
+            }
+        };
+    }
+
+    static getInstanceOrThrowError(): Recipe {
+        if (Recipe.instance !== undefined) {
+            return Recipe.instance;
+        }
+        throw new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?");
+    }
+
+    static reset() {
+        if (process.env.TEST_MODE !== "testing") {
+            throw new Error("calling testing function in non testing env");
+        }
+        Recipe.instance = undefined;
+    }
+
+    getAPIsHandled = (): APIHandled[] => {
+        return [
+            {
+                method: "get",
+                pathWithoutApiBasePath: new NormalisedURLPath(STATUS),
+                id: STATUS,
+                disabled: false,
+            },
+        ];
+    };
+
+    handleAPIRequest = async (id: string, req: BaseRequest, res: BaseResponse): Promise<boolean> => {
```

Comment:
This function also accepts `path: NormalisedURLPath, method: HTTPMethod` - where are those?